### PR TITLE
Migrate to python logging library

### LINKED ISSUE #11: Migrate to using the logging library to allow debug messages

### PR DESCRIPTION
**Pull request by V3ntus**
Restructures stdout messages to provide ease of use for debugging

Resolves #11 

658d2f4  
- Added logutil and moved initial log config/processing there

fc104ee  
- Moved chunk of code to init a logger into its own function in logutil.py

### CODE REVIEW

**Comment by savioxavier** [comment]
Thanks for this @V3ntus, the logging library is a pretty cool addition. Tested it out and it works fine.

However, I really think it would better if there were colored outputs - so as to differentiate between INFO and WARNING, as well as different types of INFOs. I'm thinking of using color escapes, but an external library would do just as well.

**Comment by V3ntus** [comment]
> Thanks for this @V3ntus, the logging library is a pretty cool addition. Tested it out and it works fine.
> 
> However, I really think it would better if there were colored outputs - so as to differentiate between INFO and WARNING, as well as different types of INFOs. I'm thinking of using color escapes, but an external library would do just as well.


I like that idea. Seems color is a simple implementation.

What sort of different types of INFOs were you considering? In debug mode, it's categorized by the function name and even line number the logging message was called from, if that's sort of close to your idea

**Comment by V3ntus** [comment]
image
INFO level vs DEBUG level

DEBUG adds two columns: the origin of the message, and the line number (in red)

File: cogs/repo.py [hunk 1]
```diff
@@ -21,10 +25,17 @@
 
 __GUILD_IDS__ = [DEV_GUILD]
 
+# Configure logging for this (main.py) handler
```
Comment (V3ntus) [inline cogs/repo.py:28]: Typo: repo.py on this comment

File: cogs/src/logutil.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class CustomFormatter(logging.Formatter):
+    grey = "\x1b[38;1m"
+    green = "\x1b[42;1m"
+    yellow = "\x1b[43;1m"
+    red = "\x1b[41;1m"
+    bold_red = "\x1b[31;1m"
+    reset = "\x1b[0m"
+
+    format = f"[%(asctime)s][%(levelname)-7s][%(name)-14s][%(lineno)4s] %(message)s"
+    FORMATS = {
```
Comment (V3ntus) [inline cogs/src/logutil.py:21]: Modify log color schemes here

**Review by savioxavier — APPROVED** [review]
Looks good to me, I'll merge!

Also, might tinker around with the colors a bit.

**Pull request merged**